### ISSUE TITLE
Cross-project code execution - ala VBA

### ISSUE THREAD

**Issue by mburns08109** [WaynePhillipsEA/twinbasic#781]
**Is your feature request related to a problem? Please describe.**
In VBA, it is possible to execute code across document boundaries, or, in software development terms - across different project scopes. I don't yet see how this will be enabled or accommodated in tB.

**Describe the solution you'd like**
In Excel VBA, you can have multiple separate "projects" loaded into the VBA environment simultaneously. In practical terms, this is simply having multiple workbooks open simultaneously. However, this has operational consequences.
Because in the Excel VBA environment, all of: a Personal.xlsm workbook; a VBA Add-in (or multiple such COM and VBA Add-ins); and multiple simultaneously-open workbooks (each with differing worksheet compositions). ALL those projects code modules are open and available to one another simultaneously.

IN VB6 terms this is functionally the same as having MyProjectA.VBP and MyProjectB.VBP both loaded in the IDE simultaneously, and having a "super-scope" where MyProjectB could call upon and execute MyProjectA's functions/subs/classes(/forms/reports, etc.).

This means, in programming terms, that Project A's code is accessible to and executable from Project B's context(s). (Indeed, this very capacity is necessary for _any_ VBA Addin code to function.)

**Describe alternatives you've considered**
none as yet.

**Additional context**
I think that describes the crux of the issue I'm trying to raise here.
I'm asking if this has been at all considered as a part of envisioning a future where tB can be called upon as a replacement IDE for VBA. If not, then this capacity is going to be a required part of trying to fulfill that role properly.
In other words - can tB be called upon in the manner like a VBA Add-In to Excel (as an architectural question)?

**Comment by WaynePhillipsEA** [comment]
Conceptually it's very similar to the _packages_ support that we already have in tB.   When tB loads a _package_, it actually loads it exactly the same as it loads an ordinary project, and then imports the public symbols into the root project.  The main difference between packages and multi-project setups is that packages are _locked_ (read-only), and this does simplify the handling somewhat since tB can just compile the package once when it loads it, and not have to worry about doing any live synchronization between multiple projects being edited.

Architecturally, I'd say we're probably 80% there already, in terms of supporting multi-projects.

**Comment by mburns08109** [comment]
I'd perhaps add to the above, in that the composition of the _package_ (to use your term), would also have to be able to automatically adjust (as does Excel VBA) for when additional workbooks are opened/closed in the "parent" application (that "hosts" VBA - presuming that like Excel and Word, they support that concept - Access does not do this, and Access VBA treats the VBA environment in a "single project only" manner).
